interop-cats/jvm/src/main/scala/scalaz/zio/interop/catszmanagedjvm.scala

```diff
@@ -0,0 +1,102 @@
+package scalaz.zio.interop
+
+import cats.effect.{ Bracket, LiftIO, Resource }
+import cats.{ ~>, effect, Bifunctor, Defer, Functor, Monad, MonadError, Monoid, Semigroup, SemigroupK }
+import scalaz.zio.{ Reservation, ZIO, ZManaged }
+
+trait CatsZManagedSyntax {
+
+  implicit class ResourceSyntax[F[_], A](resource: Resource[F, A]) {
+    def toManaged(
+      implicit nat: F ~> ZIO[Any, Throwable, ?],
+      b: Bracket[F, Throwable],
+      zB: Bracket[ZIO[Any, Throwable, ?], Throwable],
```

Comment:
You shouldn't need any constraints on `ZIO`, since you can already call all concrete methods on it directly.

Reply:
Not fixing this to throwable has the disadvantage of not being able to call orDie on the release. Which would only leave swallowing the exception. I don't know whether this is a good tradeoff.

I can rewrite the logic to not require the bracket instance Bracket[ZIO[R, E, A], Throwable] but this then instead requires a Functor[F]. Also actually calling use on the Resource would still require Bracket[ZIO[R, E, A], Throwable], for which I can't imagine a sane implementation where E isn't throwable.

Reply:
Sorry, I mean you should definitely constrain `E = Throwable`, but why require `Bracket`, etc?

Reply:
Ah ok. Yeah I can definetly do that. Those were just the requirements on mapk from cats.
I can either just copy that specifically for ZIO, or try to go from {Allocated, Bind, Suspend} directly to ZManaged. I'll try the later option first.

Reply:
I don't think we need any type class constraints on ZIO, because the type class constraints on ZIO just give us methods, however, we already have all the methods that we need directly on ZIO; and even if we want to use Cats Effects methods on ZIO, we can do that with an implicit Runtime (at most, and many Cats Effects type classes don't need that).